File: web/src/api/testPay.js
import service from '@/utils/request'

export const createPay = (data) => {
  return service({
    url: '/testPay/createPay',
    method: 'post',
    data
  })
}

export const deletePay = (data) => {
  return service({
    url: '/testPay/deletePay',
    method: 'delete',
    data
  })
}

export const deletePayByIds = (data) => {
  return service({
    url: '/testPay/deletePayByIds',
    method: 'delete',
    data
  })
}

export const updatePay = (data) => {
  return service({
    url: '/testPay/updatePay',
    method: 'put',
    data
  })
}

export const findPay = (params) => {
  return service({
    url: '/testPay/findPay',
    method: 'get',
    params
  })
}

export const getPayList = (params) => {
  return service({
    url: '/testPay/getPayList',
    method: 'get',
    params
  })
}

File: server/router/lgjx/testPay.go
package lgjx

import (
	v1 "github.com/flipped-aurora/gin-vue-admin/server/api/v1"
	"github.com/flipped-aurora/gin-vue-admin/server/middleware"
	"github.com/gin-gonic/gin"
)

type TestPayRouter struct {
}

func (s *TestPayRouter) InitPayRouter(Router *gin.RouterGroup) {
	testPayRouter := Router.Group("testPay").Use(middleware.OperationRecord())
	testPayRouterWithoutRecord := Router.Group("testPay")
	var testPayApi = v1.ApiGroupApp.LgjxTestApiGroup.TestPayApi
	{
		testPayRouter.POST("createPay", testPayApi.CreatePay)             // 新建Pay
		testPayRouter.DELETE("deletePay", testPayApi.DeletePay)           // 删除Pay
		testPayRouter.DELETE("deletePayByIds", testPayApi.DeletePayByIds) // 批量删除Pay
		testPayRouter.PUT("updatePay", testPayApi.UpdatePay)              // 更新Pay
	}
	{
		testPayRouterWithoutRecord.GET("findPay", testPayApi.FindPay)       // 根据ID获取Pay
		testPayRouterWithoutRecord.GET("getPayList", testPayApi.GetPayList) // 获取Pay列表
	}
}
